src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -16,6 +18,78 @@ def filter_notification_link(link_text, amount, filter_item)
     end
   end
 
+  def request_badge_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined, :revoke
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    else
+      'dark'
+    end
+  end
+
+  def notification_link(notification)
+    link_text = case notification.event_type
+                when 'Event::RequestStatechange', 'Event::RequestCreate', 'Event::ReviewWanted'
+                  "Request ##{notification.notifiable.number}"
+                when 'Event::CommentForRequest'
+                  "Request ##{notification.notifiable.commentable.number}"
+                when 'Event::CommentForProject'
+                  notification.notifiable.commentable.name
+                when 'Event::CommentForPackage'
+                  commentable = notification.notifiable.commentable
+                  "#{commentable.project.name} / #{commentable.name}"
+                end
+
+    link_to(link_text, notification.link_to_notification_target, class: 'mx-1 text-word-break-all')
```

Comment:
This can go do the view if `notification_link` just returns `link_text`

Reply:
We already have a `NotificationPresenter#link_to_notification_target` to get the paths we need for the link. So I'm merging these two methods to take both the link text and the link path in one shot. Have a look to `NotificationPresenter#notifiable_link`.

Reply:
Have you forgot to push this code? `NotificationPresenter` is unchanged.